### DISCUSSION TITLE
Notifications On and Off

### DISCUSSION THREAD

**Discussion by Salmankhan033** [invertase/react-native-firebase#7022] [Q&A]
I am using the firebase @react-native-firebase/messaging
How to handle On and Off the notifications from code?
I want to set a button in the app and On and off the notifications from the button.
how to handle it ?
Please help

**Comment by mikehardy** [comment]
Hi there! This is a project-specific question, so you'll have to investigate the behavior of the APIs you are using and use them the way that fits best for your project.

I can say that if you include a `notification` block in the firebase cloud message from the server, it *will* show a notification when the device receives it.

If you want a button that turns that off, what you'll need to do is implement some sort of storage for a user preference to receive notifications or not (perhaps a property in a user object stored in firestore?) and when the user turns them off, you store that they do not want notifications. Then you check that before you send the FCM.
